### ISSUE TITLE
No resultset logged when executing a stored procedure

### ISSUE THREAD

**Issue by PierreC2** [p6spy/p6spy#282]
I am trying to log the resultset from a stored procedure but nothing is logged after the procedure call.

The log file only registered the call to the DB :

    2015-03-24 22:15:37:321|11|statement|connection 0|{ CALL envoi_courriel() }|{ CALL envoi_courriel() }
    2015-03-24 22:15:37:339|0|commit|connection 0||
    2015-03-24 22:19:48:227|-1|debug|connection -1||this is com.p6spy.engine.spy.P6SpyDriver@5db99810 and passthru is com.mysql.jdbc.Driver@7f2e7127

Nothing is excluded from the logging in the spy.proerties.  That is :

    excludecategories=

Am I missing something or is this feature not supported? Thanks in advance.

**Comment by quintonm** [comment]
You have the configuration correct.  Can you post a snippet of your java code to show the method calls involved?

**Comment by quintonm** [comment]
Nevermind, I found the problem.  It is a bug.  I will have a fix for it soon.

**Comment by PierreC2** [comment]
Thank you,

This is very kind.

I will stay tuned for the next release.

Pierre C.

Envoyé de mon iPhone

**Comment by quintonm** [comment]
This has been fixed in the latest snapshot build.

**Comment by PierreC2** [comment]
Hello quintonm,

Unfortunately, the problem is not solved for me with the latest snapshot version.

I copied the new p6spy-2.1.4-SNAPSHOT.jar into  :
C:\Users\utilisateur\GlassFish_Server\glassfish\domains\domain1\lib\ext

and I still don't have any resultset log for a call to a db procedure.  Here is the pertinent excerpt from my log : 

    2015-04-12 22:59:58:485|10|statement|connection 0|{ CALL envoi_courriel() }|{ CALL envoi_courriel() }
    2015-04-12 22:59:58:500|1|commit|connection 0||

(I do have resultset logs for regular SELECT instructions), e.g. : 

    2015-04-12 22:59:26:372|0|result|connection 0|SELECT id_comite, codes_comites, date_debut, date_fin, pointage FROM comites WHERE (id_comite = ?)|
    2015-04-12 22:59:26:373|-1|resultset|connection -1|SELECT id_comite, codes_comites, date_debut, date_fin, pointage FROM comites WHERE (id_comite = ?)|1 = 1, 2 = CA, 3 = 1976-11-15, 4 = null, 5 = 100

As requested, here is the simple java code that calls the procedure :
     

    public List<NomsEtCourriels> faireEnvoiCourriel () {   
            //http://en.wikibooks.org/wiki/Java_Persistence/Advanced_Topics#Stored_Procedureshttp://en.wikibooks.org/wiki/Java_Persistence/Advanced_Topics#Stored_Procedures
            List<NomsEtCourriels> courriels;
            StoredProcedureQuery envoiCourriel = getEntityManager().createNamedStoredProcedureQuery("envoi_courriel");
            logger.info("[ejbMembresFacade]faireEnvoiCourriel envoiCourriel =" + envoiCourriel);
            courriels = envoiCourriel.getResultList();
            logger.info("[ejbMembresFacade]faireEnvoiCourriel  Nbre d'enregistrements retournés : " + courriels.size());
            if (courriels != null) {
                for (NomsEtCourriels courriel : courriels) {
                    logger.info("[ejbMembresFacade]faireEnvoiCourriel  courriel : " + courriel);
                }
           }
            return courriels;
        }

And, the entity class which is basic and only contains 3 fields :

    @Entity
    public class NomsEtCourriels implements Serializable {
    
    private Integer noLigne;
    private String noms;
    @Id
    private String courriels;
    
    
    public NomsEtCourriels() {
        System.err.println("[NomsEtCouriels]no argument constructor");
    }
    
    public NomsEtCourriels(Integer noLigne, String noms, String courriels) {
        System.err.println("[NomsEtCouriels]3 field constructor : " + noLigne + ", noms=" + noms + ", courriels=" + courriels);
        this.noLigne = noLigne;
        this.noms = noms;
        this.courriels = courriels;
    }

    + getters & setters

I want to use p6spy to diagnose why when I iterate over my resultset, only the "courriel" field is not null while, when executing the procedure call from MySql Workbench, all three fields are OK and not null.

Thank you.

**Comment by quintonm** [comment]
The problem that I found was that p6spy was not intercepting the call to CallableStatement.getResultSet() to wrap the resultset with a proxy.  I fixed that and added a test [P6TestCallableStatement.testStoredProcedureResultSet()](https://github.com/p6spy/p6spy/blob/master/src/test/java/com/p6spy/engine/spy/P6TestCallableStatement.java) to verify that it worked.  The test passes for MySql using the stored procedure shown below (copied from [liquibase.xml](https://github.com/p6spy/p6spy/blob/master/src/test/resources/liquibase.xml)).

```
create procedure test_proc_rs(cust_name varchar(100))
begin
   select * from customers where name like concat('%',cust_name,'%');
end
```

I think that one of two things is going on.
1) You might not be using the latest version of 2.1.4-SNAPSHOT.  To verify that you have the right one, open up p6spy-2.1.4-SNAPSHOT.jar and make sure that the class `com.p6spy.engine.logging.P6LogCallableStatementGetResultSetDelegate` is present.
2) Your JPA provider might not be calling the getResultSet() method of CallableStatement.  To track that down, you will need to step through the code of the JPA provider so see what it is actually doing.
